src/OpenTelemetry.AutoInstrumentation/Configuration/EnvironmentConfigurationTracerHelper.cs

```diff
@@ -63,57 +56,107 @@ public static TracerProviderBuilder UseEnvironmentVariables(this TracerProviderB
         return builder;
     }
 
-    public static TracerProviderBuilder AddSdkAspNetInstrumentation(this TracerProviderBuilder builder)
+    private static TracerProviderBuilder SetExporter(this TracerProviderBuilder builder, TracerSettings settings)
     {
+        if (settings.ConsoleExporterEnabled)
+        {
+            Wrappers.AddConsoleExporter(builder);
+        }
+
+        return settings.TracesExporter switch
+        {
+            TracesExporter.Zipkin => Wrappers.AddZipkinExporter(builder),
+            TracesExporter.Jaeger => Wrappers.AddJaegerExporter(builder),
+            TracesExporter.Otlp => Wrappers.AddOtlpExporter(builder, settings),
+            TracesExporter.None => builder,
+            _ => throw new ArgumentOutOfRangeException($"Traces exporter '{settings.TracesExporter}' is incorrect")
+        };
+    }
+
+    /// <summary>
+    /// This class wraps external extensions methods to ensure the dlls are not loaded, if not necessary.
```

Comment:
It seems worth adding a mention that preventing inlining is required.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/1530/commits/734358f34611e6883b71e159b431320518a5e3cd